File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Protocol/ProtocolUser.vb
Public Class ProtocolUser
    Inherits AuditableDomainObject(Of Guid)

    Private _protocol As Protocol

#Region " Constructor "
    Public Sub New()

    End Sub

#End Region

#Region " Properties "
    Public Overridable Property Year As Short

    Public Overridable Property Number As Integer

    Public Overridable Property Account As String

    Public Overridable Property Note As String

    Public Overridable Property Type As ProtocolUserType

    Public Overridable Property Protocol As Protocol
        Get
            Return _protocol
        End Get
        Set(ByVal value As Protocol)
            _protocol = value
            Year = value.Year
            Number = value.Number
        End Set
    End Property
#End Region

End Class

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Protocols/ProtocolUser.cs
using System;
using VecompSoftware.DocSuiteWeb.Repository.Entity;

namespace VecompSoftware.DocSuiteWeb.Entity.Protocols
{
    public class ProtocolUser : DSWBaseEntity, IUnauditableEntity
    {
        #region [ Constructor ]

        public ProtocolUser() : this(Guid.NewGuid()) { }
        public ProtocolUser(Guid uniqueId)
            : base(uniqueId) { }
        #endregion

        #region[ Properties ]
        public short Year { get; set; }
        public int Number { get; set; }
        public string Account { get; set; }
        public ProtocolUserType Type { get; set; }
        public string Note { get; set; }
        #endregion

        #region[ Navigation Properties ]

        public virtual Protocol Protocol { get; set; }

        #endregion
    }
}
